pkg/metrics/otel.go

```diff
@@ -129,7 +131,7 @@ processors:
             aggregation_type: max
       - include: declared_resources
         action: update
-        new_name: current_declared_resources
+        new_name: declared_resources
```

Comment:
Can this be a different PR?

Reply:
Sure let me modify it a bit, along with the constify